src/librbd/api/Group.cc

```diff
@@ -211,15 +213,16 @@ int group_snap_remove_by_record(librados::IoCtx& group_ioctx,
   }
 
   ret_code = 0;
-  for (int i = 0; i < snap_count; ++i) {
+  for (int i = 0; i < last_idx; ++i) {
     r = on_finishes[i]->wait();
     delete on_finishes[i];
     if (r < 0) {
       ictxs[i] = nullptr;
       ret_code = r;
     }
   }
-  if (ret_code != 0) {
+  if (ret_code != 0 || last_idx != snap_count) {
+    ldout(cct, 20) << "Removing snapshots failed" << dendl;
```

Comment:
This log message seems misleading because up to this point removing snapshots isn't even attempted -- everything that is done before this point is related to opening the images.